### PR TITLE
Fetch filter update

### PR DESCRIPTION
**Pull request by ShrijanL**
- Used get_prep_value() in fetch filter validation.
- Removed validation_func mapping dict.

### CODE REVIEW

**Review by nareshelequin — CHANGES_REQUESTED** [review]

File: django_generic_api/utils.py [hunk 1]
```diff
@@ -173,3 +136,22 @@ def is_valid_domain(domain):
...
+
+def get_field_properties(field1):
+    """
+    Retrieve field properties like 'type','null','blank',
+    'max_length', 'default'
+
+    param : Django field instance
+    return : dict with field properties
+    """
+    field_properties = {
```
Comment (nareshelequin) [inline django_generic_api/utils.py:149]: based on field type, make field properties.
ex: max_length do not have datetime field ... etc

**Review by nareshelequin — CHANGES_REQUESTED** [review]

File: django_generic_api/services.py [hunk 2]
```diff
@@ -187,24 +187,32 @@ def get_model_config_schema(model):
...
+    Verify if the field accepts null as a valid input.
+    Confirm that the value is suitable for insertion into the field.
+
+    param : model, fields, value
+    return : True/False
+    """
     is_fields_exist(model, [field1])
 
-    model_fields = get_model_fields_with_properties(model, [field1])
+    model_fields = get_model_fields_with_properties(model, field1)
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/services.py:201]: converted to [field1]

File: django_generic_api/services.py [hunk 3]
```diff
@@ -187,24 +187,32 @@ def get_model_config_schema(model):
...
     for value_i in value:
-        if not is_valid_value:
-            is_valid_value = validation_func(value_i)
-        else:
-            is_valid_value *= validation_func(value_i)
+        try:
+            field_instance.get_prep_value(value_i)
+            is_valid_value *= True
+        except (ValueError, ValidationError):
+            is_valid_value *= False
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/services.py:215]: done.

File: django_generic_api/utils.py (hunk 1 above)
Comment (ShrijanL) [reply django_generic_api/utils.py:149]: updated as 
checks if it has maxlength then gets it, else None

File: django_generic_api/utils.py [hunk 4]
```diff
@@ -42,33 +42,31 @@ def get_model_fields_with_properties(model, input_fields):
...
-    required_attributes = ["null", "blank", "max_length", "default"]
-    field_dict = {}
-    field_properties = {}
 
-    # todo: separate function to get field properties
-    for field1 in field_obj:
-        # collect properties of fields.
-        field_properties["type"] = field1.get_internal_type()
+    fields = []
+    if field_list:
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/utils.py:47]: 1. for len(field_list) > 1, ,
from filter.names , pop id and timestamp values and if len >1 , throw error

2. As the filter.names are already validated before hand, no need to validate again.

[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/utils.py:47]: 1. According to the payload model, filter.name accepts only a single string value and does not allow multiple values, so there is no need to check the length of filter.name.
If filter.name is set to id, inserted_timestamp, update_timestamp, or end_timestamp, an error will be raised indicating that the field is not permitted for filtering.

File: django_generic_api/services.py [hunk 5]
```diff
@@ -282,9 +288,26 @@ def apply_filters(model, filters):
         value = filter_item.value
         operation = filter_item.operation
 
+        UNCOUNTABLE_FIELDS = [
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/services.py:268]: removed this part.

File: django_generic_api/utils.py [hunk 6]
```diff
@@ -173,3 +135,26 @@ def is_valid_domain(domain):
...
+    'max_length', 'default'
+
+    param : Django field instance
+    return : dict with field properties
+    """
+    field_properties = {
+        "type": field1.get_internal_type(),
+        "null": field1.null,
+        "blank": field1.blank,
+        "max_length": (
```
[the remark this answers is not in the document]
Comment (ShrijanL) [reply django_generic_api/utils.py:195]: updated

**Review by nareshelequin — APPROVED** [review]

**Pull request merged**